internal/controller/virtualmachinebmc/controller.go

```diff
@@ -227,9 +357,79 @@ func (r *VirtualMachineBMCReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		return err
 	}
 
+	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &corev1.ServiceAccount{}, ownerKey, func(rawObj client.Object) []string {
+		// grab the serviceaccount object, extract the owner...
+		sa := rawObj.(*corev1.ServiceAccount)
+		owner := metav1.GetControllerOf(sa)
+		if owner == nil {
+			return nil
+		}
+		// ...make sure it's a VirtualMachineBMC...
+		if owner.APIVersion != apiGVStr || owner.Kind != "VirtualMachineBMC" {
+			return nil
+		}
+
+		// ...and if so, return it
+		return []string{owner.Name}
+	}); err != nil {
+		return err
+	}
+
+	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &rbacv1.RoleBinding{}, ownerKey, func(rawObj client.Object) []string {
+		// grab the rolebinding object, extract the owner...
+		rb := rawObj.(*rbacv1.RoleBinding)
+		owner := metav1.GetControllerOf(rb)
+		if owner == nil {
+			return nil
+		}
+		// ...make sure it's a VirtualMachineBMC...
+		if owner.APIVersion != apiGVStr || owner.Kind != "VirtualMachineBMC" {
+			return nil
+		}
+
+		// ...and if so, return it
+		return []string{owner.Name}
+	}); err != nil {
+		return err
+	}
+
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&virtualmachinev1.VirtualMachineBMC{}).
+		For(&bmcv1.VirtualMachineBMC{}).
 		Owns(&corev1.Pod{}).
 		Owns(&corev1.Service{}).
+		Owns(&corev1.ServiceAccount{}).
+		Owns(&rbacv1.RoleBinding{}).
+		Watches(
+			&kubevirtv1.VirtualMachine{},
+			handler.EnqueueRequestsFromMapFunc(r.findVirtualMachineBMCsForVM),
+		).
```

Comment:
I'm sure this can benefit us someday, but could you elaborate on which part currently needs the VirtualMachineBMC reconciled upon VirtualMachine changes?